qiskit/aqua/components/neural_networks/quantum_generator.py

```diff
@@ -349,9 +349,9 @@ def train(self, quantum_instance=None, shots=None):
         """
 
         self._shots = shots
-        # Force single optimization iteration
-        self._optimizer._maxiter = 1
-        self._optimizer._t = 0
+        if self._optimizer._maxiter != 1:
```

Comment:
Currently there is no standard interface to query this. Some optimizers take max fun as well as max iter type settings, where the former, without deeper knowledge of what is happening internal figuring iters from funs is problematic.

I would suggest this type of text could just be the docstring for the optimizer param as I am not sure there is any real way to check this across the optimizers. So if a user supplies their own optimizer blindly without reading the text then yes things might behave unexpectedly - but hopefully they would consult the documentation.

Reply:
In thinking since maxiter is placed in the options dictionary that could be checked and if the key exists then its value could be checked. If the key did not exist or its value was not 1 then an appropriate warning could be given. The options dictionary is not exposed via a public getter interface but could be (it can be accessed via its private instance variable, but if this is something that was wanted maybe making a getter and having it public would be better). This would be a simple change to base optimizer class to make a getter is you feel that this level of checking is needed and just documenting it will not suffice.